src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -0,0 +1,233 @@
+
+  private AtomicInteger correlationId = new AtomicInteger(0);
+
+  private final Decoder<Object> avroDecoder;
+  private final Decoder<byte[]> binaryDecoder;
+
+  public SimpleConsumerManager(final KafkaRestConfig config,
+                               final MetadataObserver mdObserver,
+                               final SimpleConsumerFactory simpleConsumerFactory) {
+
+    this.mdObserver = mdObserver;
+    this.simpleConsumerFactory = simpleConsumerFactory;
+
+    maxPoolSize = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_SIZE_CONFIG);
+    simpleConsumersPools = new HashMap<Broker, SimpleConsumerPool>();
+
+    // Load decoders
+    Properties props = new Properties();
+    props.setProperty("schema.registry.url",
+        config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
+    avroDecoder = new KafkaAvroDecoder(new VerifiableProperties(props));
+
+    binaryDecoder = new DefaultDecoder(new VerifiableProperties());
+  }
+
+  private SimpleConsumerPool createSimpleConsumerPool() {
+    if (maxPoolSize > 0) {
+      return new SizeLimitedSimpleConsumerPool(maxPoolSize, simpleConsumerFactory);
+    }
+    return new NaiveSimpleConsumerPool(simpleConsumerFactory);
+  }
+
+  private SimpleFetcher getSimpleFetcher(final Broker broker) {
+    if (!simpleConsumersPools.containsKey(broker)) {
+      simpleConsumersPools.put(broker, createSimpleConsumerPool());
+    }
+    final SimpleConsumerPool pool = simpleConsumersPools.get(broker);
+    return pool.get(broker.host(), broker.port());
+  }
+
+  public void consume(final String topicName,
+                      final int partitionId,
+                      long offset,
+                      long count,
+                      final EmbeddedFormat embeddedFormat,
+                      final ConsumerManager.ReadCallback callback) {
+
+    List<ConsumerRecord> records = null;
+    Exception exception = null;
+    SimpleFetcher simpleFetcher = null;
+
+    try {
+      final Broker broker = mdObserver.getLeader(topicName, partitionId);
+      simpleFetcher = getSimpleFetcher(broker);
+
+      records = new ArrayList<ConsumerRecord>();
+
+      int fetchIterations = 0;
+      while (count > 0) {
+        fetchIterations++;
+        log.debug("Simple consumer " + simpleFetcher.clientId() +
+            ": fetch " + fetchIterations + "; " + count + " messages remaining");
+
+        final ByteBufferMessageSet messageAndOffsets =
+            fetchRecords(topicName, partitionId, offset, simpleFetcher);
+
+        for (final MessageAndOffset messageAndOffset : messageAndOffsets) {
+          records.add(createConsumerRecord(messageAndOffset, topicName, partitionId, embeddedFormat));
+          count--;
+          offset++;
+
+          // If all requested messages have already been fetched, we can break early
+          if (count == 0) {
+            break;
+          }
+        }
+      }
+
+    } catch (RestException e) {
```

Comment:
mdObserver.getLeader() could throw a RestNotFoundException if the the leader is not found.
fetchRecords() could throw a RestServerErrorException if the response contains a kafka error.

Do you mean I should put comments here to list the exceptions that can be thrown ?